src/renderer/components/+catalog/catalog-entity-details.tsx

```diff
@@ -27,32 +27,34 @@ import type { CatalogCategory, CatalogEntity } from "../../../common/catalog";
 import { Icon } from "../icon";
 import { CatalogEntityDrawerMenu } from "./catalog-entity-drawer-menu";
 import { CatalogEntityDetailRegistry } from "../../../extensions/registries";
-import type { CatalogEntityItem } from "./catalog-entity-item";
 import { isDevelopment } from "../../../common/vars";
 import { cssNames } from "../../utils";
 import { Avatar } from "../avatar";
+import { getIconColourHash } from "../../../common/catalog/helpers";
+import { EntityIcon } from "../entity-icon";
+import { getLabelBadges } from "./helpers";
 
-interface Props<T extends CatalogEntity> {
-  item: CatalogEntityItem<T> | null | undefined;
+interface Props {
+  entity: CatalogEntity | null | undefined;
   hideDetails(): void;
+  onRun: () => void;
 }
 
 @observer
-export class CatalogEntityDetails<T extends CatalogEntity> extends Component<Props<T>> {
+export class CatalogEntityDetails extends Component<Props> {
   categoryIcon(category: CatalogCategory) {
-    if (category.metadata.icon.includes("<svg")) {
-      return <Icon svg={category.metadata.icon} smallest />;
-    } else {
-      return <Icon material={category.metadata.icon} smallest />;
-    }
+    const key = category.metadata.icon.includes("<svg") ? "svg" : "material";
+
+    return <Icon {...{ [key]: category.metadata.icon }} smallest />;
   }
 
-  renderContent(item: CatalogEntityItem<T>) {
-    const detailItems = CatalogEntityDetailRegistry.getInstance().getItemsForKind(item.kind, item.apiVersion);
-    const details = detailItems.map(({ components }, index) => {
-      return <components.Details entity={item.entity} key={index}/>;
-    });
+  renderContent(entity: CatalogEntity | undefined, onRun: () => void) {
```

Comment:
The other change that was required by removing `CatalogEntityItem` was the injection of the `onRun` function here because it needs to go through the `catalogEntityRegistry` but so as to keep it injectable, I did not import that.